File: app/src/main/java/com/tokeninc/sardis/application_template/helpers/PrintHelpers/BasePrintHelper.java
package com.tokeninc.sardis.application_template.helpers.PrintHelpers;
/*
import com.token.printerlib.PrinterDefinitions;
import com.token.printerlib.StyledString;
import com.tokeninc.sardis.application_template.Entity.SampleReceipt;

public class BasePrintHelper {

    static void addText(StyledString styledText, String text, PrinterDefinitions.Alignment alignment) {
        addText(styledText, text, alignment, 11, 0);
    }

    static void addText(StyledString styledText, String text, PrinterDefinitions.Alignment alignment, int fontSize, float lineSpacing) {
        styledText.setFontFace(PrinterDefinitions.Font_E.SourceCodePro);
        styledText.setFontSize(fontSize);
        styledText.setLineSpacing(lineSpacing);
        styledText.addTextToLine(text, alignment);
    }

    static void addTextToNewLine(StyledString styledText, String text, PrinterDefinitions.Alignment alignment) {
        styledText.newLine();
        addText(styledText, text, alignment);
    }

    static void addTextToNewLine(StyledString styledText, String text, PrinterDefinitions.Alignment alignment, int fontSize) {
        styledText.newLine();
        addText(styledText, text, alignment, fontSize, 0);
    }

    static void printSlipHeader(StyledString styledText, SampleReceipt receipt){
        styledText.setLineSpacing(0.5f);
        styledText.setFontSize(12);
        styledText.setFontFace(PrinterDefinitions.Font_E.SourceSansPro);
        styledText.addTextToLine(receipt.getMerchantName(), PrinterDefinitions.Alignment.Center);

        styledText.newLine();
        styledText.setFontFace(PrinterDefinitions.Font_E.Sans_Semi_Bold);
        styledText.addTextToLine("İŞYERİ NO:", PrinterDefinitions.Alignment.Left);
        styledText.setFontFace(PrinterDefinitions.Font_E.SourceSansPro);
        styledText.addTextToLine(receipt.getMerchantID(), PrinterDefinitions.Alignment.Right);

        styledText.newLine();
        styledText.setFontFace(PrinterDefinitions.Font_E.Sans_Semi_Bold);
        styledText.addTextToLine("TERMİNAL NO:", PrinterDefinitions.Alignment.Left);
        styledText.setFontFace(PrinterDefinitions.Font_E.SourceSansPro);
        styledText.addTextToLine(receipt.getPosID(), PrinterDefinitions.Alignment.Right);
    }

}


 */

File: app/src/main/java/com/tokeninc/sardis/application_template/helpers/printHelpers/BasePrintHelper.kt
package com.tokeninc.sardis.application_template.helpers.printHelpers

import android.content.Context
import com.token.printerlib.PrinterDefinitions
import com.token.printerlib.StyledString
import com.tokeninc.sardis.application_template.MainActivity
import com.tokeninc.sardis.application_template.R
import com.tokeninc.sardis.application_template.SampleReceipt

open class BasePrintHelper {

    fun addText(styledText: StyledString, text: String?, alignment: PrinterDefinitions.Alignment?) {
        addText(styledText, text, alignment, 11, 0f)
    }

    fun addText(styledText: StyledString, text: String?, alignment: PrinterDefinitions.Alignment?,
        fontSize: Int, lineSpacing: Float
    ) {
        styledText.setFontFace(PrinterDefinitions.Font_E.SourceCodePro)
        styledText.setFontSize(fontSize)
        styledText.setLineSpacing(lineSpacing)
        styledText.addTextToLine(text, alignment)
    }

    fun addTextToNewLine(styledText: StyledString, text: String?, alignment: PrinterDefinitions.Alignment?
    ) {
        styledText.newLine()
        addText(styledText, text, alignment)
    }

    fun addTextToNewLine(styledText: StyledString, text: String?,
        alignment: PrinterDefinitions.Alignment?, fontSize: Int
    ) {
        styledText.newLine()
        addText(styledText, text, alignment, fontSize, 0f)
    }

    fun printSlipHeader(styledText: StyledString, mainActivity: MainActivity) {
        styledText.setLineSpacing(0.5f)
        styledText.setFontSize(12)
        styledText.setFontFace(PrinterDefinitions.Font_E.SourceSansPro)
        styledText.addTextToLine(mainActivity.getString(R.string.merchant_name), PrinterDefinitions.Alignment.Center)
        styledText.newLine()
        styledText.newLine()
        styledText.newLine()
        styledText.setFontFace(PrinterDefinitions.Font_E.Sans_Semi_Bold)
        styledText.addTextToLine("İŞYERİ NO:", PrinterDefinitions.Alignment.Left)
        styledText.setFontFace(PrinterDefinitions.Font_E.SourceSansPro)
        styledText.addTextToLine(mainActivity.activationViewModel.getMerchantId(), PrinterDefinitions.Alignment.Right)
        styledText.newLine()
        styledText.setFontFace(PrinterDefinitions.Font_E.Sans_Semi_Bold)
        styledText.addTextToLine("TERMİNAL NO:", PrinterDefinitions.Alignment.Left)
        styledText.setFontFace(PrinterDefinitions.Font_E.SourceSansPro)
        styledText.addTextToLine(mainActivity.activationViewModel.getTerminalId(), PrinterDefinitions.Alignment.Right)
    }

}